sips/sip-015/sip-015-network-upgrade.md

```diff
@@ -0,0 +1,1595 @@
+  relevant functions are `stack-stx`, `delegate-stack-stx`,
+  `stack-aggregation-commit`, and `reject-pox`.
+
+### New method: `stack-extend`
+
+This method allows direct stackers to re-lock their funds for up to an additional
+12 cycles *before* their current lock has expired. The target use case for
+this is to allow users to repeatedly stack without a cooldown phase.
+
+This method checks that the stacker is still allowed to stack, that
+the stacker has not delegated to an operator, and that their funds are
+currently stacked. The caller may supply a new reward address for the
+extension.
+
+The special case handler for the PoX contract in the Clarity VM will
+check this method's return value and set the stacker's STX account to
+"auto-unlock" at the end of the last extended-to reward cycle.
+
+### New method: `delegate-stack-extend`
+
+This method allows operators to re-lock the funds of one of their
+delegation participants for additional cycles *before* that participants
+lock has expired. The target use case for this is to allow delegation
+operators to repeatedly stack on behalf of their users without cooldowns.
+
+This method checks that the delegator is still authorized on behalf of
+the given stacker (and remains authorized until the end of the
+locked-for period) and that the funds are currently stacked.
+
+Note that, just as with the existing `delegate-stack-stx` function,
+this method locks *but does not commit* the user's STX. The delegation
+operator must invoke `stack-aggregation-commit` to set a reward address
+for the locked funds.
+
+The special case handler for the PoX contract in the Clarity VM will
+check this method's return value and set the stacker's STX account to
+"auto-unlock" at the end of the last extended-to reward cycle.
+
+### New method: `stack-increase`
+
+This method allows direct stackers to lock additional funds for
+the remaining cycles on their current lock.
+
+This method checks that the stacker is still allowed to stack, that
+the stacker has not delegated to an operator, that their funds are
+currently stacked, and that they have enough unlocked funds to cover
+the increase. The caller may *not* supply a new reward address for the
+increase.
+
+The special case handler for the PoX contract in the Clarity VM will
+check this method's return value and set the locked amount in the
+stacker's STX account to correspond to the increased amount.
+
+### New method: `delegate-stack-increase`
+
+This method allows operators to lock additional funds for one of their
+delegation participants for the remaining cycles on that user's
+current lock.
+
+This method checks that the delegator is still authorized on behalf of
+the given stacker (and remains authorized until the end of the
+locked-for period), that the increased amount remains less than the
+delegation's `amount-ustx` field, that the user is currently locked,
+and that the user has enough unlocked funds to cover the increase.
```

Comment:
Ideally, the delegate-stack-increase function for delegators allows to specify a new reward address as well.

Reply:
I will look into how feasible this is and get back to you early next week.

Reply:
Any updates here? @jcnelson

Reply:
Actually, this is already supported.  Looking at the code for `pox-2`, the user can already call `revoke-delegate-stx` and then `delegate-stx` to increase their lockup allowance.  After this, the pool operator can call `delegate-stack-increase` with the new allowance.  I forgot to add this to the SIP -- `delegate-stx` has already been modified in `pox-2` so that the user can call it while their STX are locked.